azext_iot/deviceupdate/params.py

```diff
@@ -305,7 +325,13 @@ def load_deviceupdate_arguments(self, _):
         )
 
     with self.argument_context("iot device-update device deployment") as context:
-        context.argument("deployment_id", options_list=["--deployment-id", "--did"], help="Deployment Id.")
+        context.argument(
+            "deployment_id",
+            options_list=["--deployment-id", "--did"],
+            help="The caller-provided deployment Id. This cannot be longer than 73 characters, "
+            "must be all lower-case, and cannot contain '&', '^', '[', ']', '{', '}', '|', '<', '>', "
```

Comment:
Is this list of characters taken from the official API documentation? Are all the other special characters supported?

Reply:
Yah that is right from the official documentation.

Reply:
Honestly I haven't tried every other special character not in that list, but I'd like to think API documentation would be accurate and if something was missed theoretically the API should error (and with luck we get an issue in the repo requesting doc improvement). In later refreshes we could spot check for additional API doc updates.